src/utils/ammUtils.ts

```diff
@@ -37,10 +55,81 @@ export const getPenaltyPercentage = (
   const output = outputAmount.simple
   const penalty = output / inputAmount.simple
 
-  if (output < min || output > max) {
-    if (reverse) {
-      return penalty > 1 ? (penalty - 1) * -100 : (1 - penalty) * 100
-    }
-    return penalty > 1 ? (penalty - 1) * 100 : (1 - penalty) * -100
+  // if (output < min || output > max) {
+  if (reverse) {
+    return penalty > 1 ? (penalty - 1) * -100 : (1 - penalty) * 100
+  }
+  return penalty > 1 ? (penalty - 1) * 100 : (1 - penalty) * -100
+  // }
+}
+
+// Scale asset via ratio
+export const useScaleAsset = (): {
```

Comment:
I'm not convinced that this is the right pattern... I think we can probably come up with something simpler by adjusting the data layer a bit. I might just have a dig into this.